Expand further minimization section of documentation (#7350)
bytecodealliance/wasmtime: docs/examples-minimal.md

## Minimizing further

Above shows an example of taking the default `cargo build` result of 130M down
to a 2.1M binary for the `wasmtime` executable. Similar steps can be done to
reduce the size of the C API binary artifact as well which currently produces a

> ~2.8M dynamic library. This is currently the smallest that can be gained with
> the source-code as-is, but there's still possible size reductions which haven't
> been implemented yet.

Reviewer: ```suggestion
~2.8M dynamic library. This is currently the smallest size with the source code
as-is, but there are more size reductions which haven't been implemented yet.
```

  strings for error messages and other purposes throughout the implementation.
  Most of this is intended for debugging and understanding more when something
  goes wrong, but much of this is not necessary for a truly minimal embedding.
  In theory much of this could be conditionally compiled out of the Wasmtime
  project to produce a smaller executable. Just how much of the final binary
  size is accounted for by formatting string is unknown, but it's well known in
  Rust that `std::fmt` is not the slimmest of modules.

* Cranelift vs Winch - the "min" builds on CI try to exclude Cranelift from
  their binary footprint (e.g. the CLI excludes it) but this comes at a cost of
  the final executable not supporting compilation of wasm modules. If this is
  required then no effort has yet been put into minimizing the code size of
  Cranelift itself. One possible tradeoff that can be made though is to choose
  between the Winch baseline compiler vs Cranelift. Winch should be much smaller
  from a compiled footprint point of view while not sacrificing everything in
  terms of performance. Note though that Winch is still under development.

> Above is some future avenues to take in terms of reducing the binary size of

Reviewer: ```suggestion
Above are some future avenues to take in terms of reducing the binary size of
```
